openvpn/fetch-external-configs.sh

```diff
@@ -6,81 +6,41 @@ set -o pipefail
-  curl -sSL --fail -o "${config_repo_temp_zip_file}" "${GITHUB_CONFIG_BUNDLE_URL}"
-
-  # Create a temporary folder and extract configs there
-  echo "Extracting configs to ${config_repo_temp_dir}"
-  unzip -q "${config_repo_temp_zip_file}" -d "${config_repo_temp_dir}"
-
-  # Find the specified provider folder. Should be under <tmpDir>/<some-root-folder>/openvpn/<provider>
-  provider_configs=$(find "${config_repo_temp_dir}"/*/openvpn -type d -name "${VPN_PROVIDER}")
-  if [[ -z "${provider_configs}" ]]; then
-    echo "ERROR: Could not find any configs for provider ${VPN_PROVIDER^^} in downloaded configs"
-    exit 1
-  fi
-
-  # Replace current provider home folder with the downloaded directory
-  echo "Found configs for ${VPN_PROVIDER^^} in ${provider_configs}, will replace current content in ${VPN_PROVIDER_HOME}"
-  rm -rf "${VPN_PROVIDER_HOME}"
-  mv "${provider_configs}" "${VPN_PROVIDER_HOME}"
-
-  exit 0
+# Add safe directory for repo folder
+git config --global --add safe.directory "${config_repo}"
 
-elif [[ "${VPN_CONFIG_SOURCE_TYPE}" == "github_clone" ]]; then
-  GITHUB_CONFIG_REPO_URL="https://github.com/${GITHUB_CONFIG_SOURCE_REPO}.git"
-  config_repo=/tmp/config-repo
-
-  echo "Will get configs from ${GITHUB_CONFIG_REPO_URL}"
-  # Check if git repo exists and clone or pull based on that
-  if [[ -d ${config_repo} ]]; then
+echo "Will get configs from ${GITHUB_CONFIG_REPO_URL}"
+# Check if git repo exists and clone or pull based on that
+if [[ -d ${config_repo} ]]; then
+  GITHUB_CONFIG_SOURCE_LOCAL=$(git -C "${config_repo}" remote -v | head -1 | awk '{print $2}' | sed -e 's/https:\/\/github.com\///' -e 's/.git//')
+  if [ "$GITHUB_CONFIG_SOURCE_LOCAL" == "$GITHUB_CONFIG_SOURCE_REPO" ]; then
     echo "Repository is already cloned, checking for update"
-    cd ${config_repo}
-    git pull
-    git checkout "${GITHUB_CONFIG_SOURCE_REVISION}"
+    git -C "${config_repo}" pull
+    git -C "${config_repo}" checkout "${GITHUB_CONFIG_SOURCE_REVISION}"
   else
     echo "Cloning ${GITHUB_CONFIG_REPO_URL} into ${config_repo}"
-    git clone "${GITHUB_CONFIG_REPO_URL}" ${config_repo}
-    cd ${config_repo}
-    git checkout "${GITHUB_CONFIG_SOURCE_REVISION}"
-  fi
-
-  # Find the specified provider folder. Should be under <tmpDir>/<some-root-folder>/openvpn/<provider>
-  provider_configs=$(find "${config_repo}"/openvpn -type d -name "${VPN_PROVIDER}")
-  if [[ -z "${provider_configs}" ]]; then
-    echo "ERROR: Could not find any configs for provider ${VPN_PROVIDER^^} in downloaded configs"
-    exit 1
-  fi
-
-  # Replace current provider home folder with the one we've fetched
-  echo "Found configs for ${VPN_PROVIDER^^} in ${provider_configs}, will replace current content in ${VPN_PROVIDER_HOME}"
-  rm -r "${VPN_PROVIDER_HOME}"
-  cp -r "${provider_configs}" "${VPN_PROVIDER_HOME}"
-
-  exit 0
-
+    config_repo_old="${config_repo}" + "_old"
+    mv "${config_repo}" "${config_repo_old}"
```

Comment:
Don't see why we want to keep a backup of the old one. Can always clone the previous repo again. Could just delete it and clone again.